crates/bevy_reflect/src/lib.rs

```diff
@@ -46,6 +46,30 @@ pub use type_uuid::*;
 pub use bevy_reflect_derive::*;
 pub use erased_serde;
 
+#[doc(hidden)]
+pub mod __macro_exports {
+    use crate::Uuid;
+
+    #[allow(clippy::unusual_byte_groupings)] // unusual byte grouping is meant to signal the relevant bits
+    pub const fn generate_composite_uuid(a: Uuid, b: Uuid) -> Uuid {
+        let mut new = [0; 16];
+        let mut i = 0;
+        while i < new.len() {
+            new[i] = a.as_bytes()[i] ^ b.as_bytes()[i];
+
+            i += 1;
+        }
+
+        // Version: the most significant 4 bits in the 6th byte: 11110000
```

Comment:
Can / should we duplicate these comments on the doc strings for this function itself?

Reply:
I added a comment to the function:
```rust
    /// Generates a new UUID from the given UUIDs `a` and `b`,
    /// where the bytes are generated by a bitwise `a ^ b.rotate_right(1)`.
    /// The generated UUID will be a UUIDv4 (meaning that the bytes should be random, not e.g. derived from the system time).
```